### PR TITLE
Subdir images 

### LINKED ISSUE #3073: Error when upload file to Rocket Chat in subfolder
Your Rocket.Chat version: 0.28 (make sure you are running the latest)

Hi,
I have some problem with Rocket Chat in subfolder
- I can chat via web but can't access via App on Mobile
- When I upload any file, it can't display with Image, text,.. and I can't download it too.
Can you help me resolv them ?

This is error when i upload a photo.

### PR DESCRIPTION
**Pull request by AlecTroemel**
@RocketChat/core 

Closes #3073 

Fix issue where images cannot be found while rocketchat being hosted behind reverse proxy. I wrote a more detailed explanation of the fix in the issue discussion.

### CODE REVIEW

File: lib/fileUpload.coffee [hunk 1]
```diff
@@ -13,7 +13,14 @@ if UploadFS?
...
 			document.cookie = 'rc_uid=' + escape(Meteor.userId()) + '; path=/'
-			document.cookie = 'rc_token=' + escape(Meteor._localStorage.getItem('Meteor.loginToken')) + '; path=/'
+			pathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX
+			loginTokenKey = 'Meteor.loginToken'
+
+			# account for if being run in subdirectory
+			if pathPrefix
+				document.cookie = 'rc_token=' + escape(Meteor._localStorage.getItem(loginTokenKey + ':/:' + pathPrefix)) + '; path=/'
+			else
+				document.cookie = 'rc_token=' + escape(Meteor._localStorage.getItem(loginTokenKey)) + '; path=/'
```
Comment (rodrigok) [inline lib/fileUpload.coffee]: We can improve all this part using `Accounts._storedLoginToken()` to get the token, what do you think? This method uses `Accounts.LOGIN_TOKEN_KEY` that already have the correct key, so seems to be the best way to do that.

Comment (AlecTroemel) [inline lib/fileUpload.coffee]: Just confirmed that that works, definitely the simpler solution, pushing change

File: lib/fileUpload.coffee [hunk 2]
```diff
@@ -13,8 +13,8 @@ if UploadFS?
 		cookie = new Cookies()
 		if Meteor.isClient
 			document.cookie = 'rc_uid=' + escape(Meteor.userId()) + '; path=/'
-			document.cookie = 'rc_token=' + escape(Meteor._localStorage.getItem('Meteor.loginToken')) + '; path=/'
-
+			document.cookie = 'rc_token=' + Accounts._storedLoginToken()
```
Comment (rodrigok) [inline lib/fileUpload.coffee]: Should we keep the `escape` method?

Comment (AlecTroemel) [inline lib/fileUpload.coffee]: I'm unsure of the functionality of escape. Is it the (deprecated) javascript escape function, underscore.js's escape function, or something else? Either way it seems right to follow the format of what was there before :) (though it didn't seem to make a difference for me).

**Comment by rodrigok** [comment]
Awesome, thanks

**Pull request merged**